routes/auth-routes.js

```diff
@@ -7,21 +7,24 @@
  */
 var authToken = require('../app/auth-token');
 var config = require('../app/config');
+var link = require('../app/public-link');
 var upload = require('../app/notebook-upload');
 var router = require('express').Router();
 var urljoin = require('url-join');
 
-var GET_URL = urljoin(config.get('PUBLIC_LINK'), '/dashboards');
 var UPLOAD_MESSAGE = 'Notebook successfully uploaded';
 
 /* POST /notebooks/* - upload a dashboard notebook */
 router.post('/notebooks(/*)', authToken, upload, function(req, res) {
+    var publicLink = link(req);
```

Comment:
Did you try this with the end-to-end docker compose? My gut tells me this will compute the wrong value because it's the bundler hitting the URL with its Host header (some internal private docker container name) instead of a user hitting it from the outside world with a browser.

I wonder if this needs to be computed once on first user visit to any user-facing dashboard server route and saved off. That's when the Host header will be correct for users visiting the server and that's the type of URL what we want to include in the POST response, something a user can be redirected to.

Yes, there's a problem with the case where the POST to notebooks happens before a user visits the server.